### ISSUE TITLE
Pipeline failed to restart though it couldn't connect to and create the consumergroup on kafka

### ISSUE THREAD

**Issue by louisnow** [numaproj/numaflow#725]
**Describe the bug**
Pipeline failed to restart though it couldn't connect to and create the consumer group on kafka.
We have 4 pipelines, each with their own kafka topic source vertex.
Out of 4 pipelines, three restarted till they successfully connected but one stayed up despite being unable to restart at source.

**To Reproduce**

1. Our kafka/zookeeper instance went down
2. One of the source vertexes received a `kafka server: Replication-factor is invalid.` error but continued staying up instead of restarting. 

For some background, we scale down our clusters and our nodes when not in use during off business hours and scale it back up when needed to reduce our infra cost.

**Expected behavior**
- Pipelines Source vertex ideally restarts until success instead of staying up unsuccessfully

**Screenshots**

**Environment (please complete the following information):**

- Kubernetes: [e.g. v1.24]
- Numaflow: [e.g. v0.8.0]

**Additional context**
```
"numaflow.Source-processor","caller":"kafka/reader.go:394","msg":"Initialization of consumer failed with error: ","vertex":"atlan-nf-pipe-webhooks-kafka-source","error":"kafka server: Replication-factor is invalid."}

{"level":"info","ts":1683256265.8245962,"logger":"numaflow.Source-processor","caller":"fetch/processor_manager.go:145","msg":"Processor has been inactive for 10 heartbeats, deleting...","vertex":"atlan-nf-pipe-webhooks-kafka-source","key":"atlan-nf-pipe-webhooks-kafka-source-1","atlan-nf-pipe-webhooks-kafka-source-1":"atlan-nf-pipe-webhooks-kafka-source-1 status:active, timeline: [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1] -> [-1:-1]"}
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
@louisnow - do you mean 3 of the pipeline sources restarted automatically and got back running, but one of them didn't?

**Comment by whynowy** [comment]
> @louisnow - do you mean 3 of the pipeline sources restarted automatically and got back running, but one of them didn't?

And I assume manually killing the pod to get it restarted fixed the problem?

**Comment by louisnow** [comment]
>  do you mean 3 of the pipeline sources restarted automatically and got back running, but one of them didn't?

- You're right. For more context, we have 4 pipelines. Each pipeline reads from a kafka topic with their own consumer group.
- Which gives up a total of 4 source vertexes from kafka, one in each pipeline.
- Three of them restarted about 7 times until they recovered but one of the source vertexes restarted 3 times and continued to stay up even though it shouldn't have.

**Comment by whynowy** [comment]
> > do you mean 3 of the pipeline sources restarted automatically and got back running, but one of them didn't?
> 
> * You're right. For more context, we have 4 pipelines. Each pipeline reads from a kafka topic with their own consumer group.
> * Which gives up a total of 4 source vertexes from kafka, one in each pipeline.
> * Three of them restarted about 7 times until they recovered but one of the source vertexes restarted 3 times and continued to stay up even though it shouldn't have.

`restarted 3 times` means the container restart? If it's possible, could you post the logs before those restarts? Based on our [code logic](https://github.com/numaproj/numaflow/blob/main/pkg/sources/kafka/reader.go#L394), the container would not crash and restart, so I want to figure out what happened.

> * A restart fixed the problem

**Comment by louisnow** [comment]
Here's the logs from the container that restarted
```
{"level":"info","ts":1683256177.144862,"logger":"numaflow.Source-processor","caller":"kafka/reader.go:381","msg":"creating NewConsumerGroup","vertex":"atlan-nf-pipe-webhooks-kafka-source","topic":"ATLAS_ENTITIES","consumerGroupName":"nf-webhooks","brokers":["kafka-headless.kafka.svc.cluster.local:9092"]}
{"level":"panic","ts":1683256177.1449268,"logger":"numaflow.Source-processor","caller":"kafka/reader.go:383","msg":"Problem initializing sarama client","vertex":"atlan-nf-pipe-webhooks-kafka-source","error":"kafka: client has run out of available brokers to talk to (Is your cluster reachable?)","stacktrace":"github.com/numaproj/numaflow/pkg/sources/kafka.(*KafkaSource).startConsumer\n\t/home/runner/work/numaflow/numaflow/pkg/sources/kafka/reader.go:383"}
panic: Problem initializing sarama client

goroutine 364 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0005460c0, {0xc000050d40, 0x1, 0x1})
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.19.1/zapcore/entry.go:232 +0x44c
go.uber.org/zap.(*SugaredLogger).log(0xc000685db8, 0x4, {0x21729da?, 0x19?}, {0x0?, 0x0?, 0x0?}, {0xc000929db8, 0x1, 0x1})
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.19.1/sugar.go:227 +0xee
go.uber.org/zap.(*SugaredLogger).Panicw(...)
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.19.1/sugar.go:204
github.com/numaproj/numaflow/pkg/sources/kafka.(*KafkaSource).startConsumer(0xc0001c41e0)
	/home/runner/work/numaflow/numaflow/pkg/sources/kafka/reader.go:383 +0x48d
created by github.com/numaproj/numaflow/pkg/sources/kafka.(*KafkaSource).Start
	/home/runner/work/numaflow/numaflow/pkg/sources/kafka/reader.go:199 +0x56
```

**Comment by louisnow** [comment]
Thanks for the quick fix @whynowy ! Do you think there could be a release with an incremental bug fix as 0.8.1 ? 
I can see master has quite a few changes right now likely in preparation for 0.9
